model/task_lifecycle.go

```diff
@@ -1143,27 +1184,28 @@ func evalStepback(ctx context.Context, t *task.Task, caller, status string, deac
 				return errors.Errorf("no tasks in task group '%s' for task '%s'", t.TaskGroup, t.Id)
 			}
 			for _, tgTask := range tasks {
-				catcher.Wrapf(doStepback(ctx, &tgTask), "stepping back task group task '%s'", tgTask.DisplayName)
+				catcher.Wrapf(doLienarStepback(ctx, &tgTask), "stepping back task group task '%s'", tgTask.DisplayName)
 				if tgTask.Id == t.Id {
 					break // don't need to stepback later tasks in the group
 				}
 			}
 
 			return catcher.Resolve()
 		}
-		return errors.Wrap(doStepback(ctx, t), "performing stepback")
-
+		return errors.Wrap(doLienarStepback(ctx, t), "performing linear stepback")
 	} else if status == evergreen.TaskSucceeded && deactivatePrevious && t.Requester == evergreen.RepotrackerVersionRequester {
-		// if the task was successful and is a mainline commit (not git tag or project trigger),
-		// ignore running previous activated tasks for this buildvariant
-		if err := DeactivatePreviousTasks(ctx, t, caller); err != nil {
-			return errors.Wrap(err, "deactivating previous task")
-		}
+		// When stepback finishes (the task is successful), and stepback was done on mainline commits,
+		// ignore running previous activated tasks for this build variant.
+		return errors.Wrap(DeactivatePreviousTasks(ctx, t, caller), "deactivating previous tasks")
 	}
-
 	return nil
 }
 
+// evalBisectStepback performs bisect stepback on the task. To be implemented in EVG-20788.
```

Comment:
You can remove the "To be implemented" sentence and add a `TODO: EVG-20788 implement stepback bisection` right before the error statement